From 97fb4f822dddaea6947862cb2d0a75e3b72b707c Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Tue, 12 Apr 2011 18:27:36 +0300
Subject: [PATCH 34/75] use system_hlid if ap is not up yet

someone tries to send CTS before ap is up. use system_hlid in this case.

TODO: understand it better. currently, it's just a workaround.
---
 drivers/net/wireless/wl12xx/tx.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/tx.c b/drivers/net/wireless/wl12xx/tx.c
index fa2d082..d4af833 100644
--- a/drivers/net/wireless/wl12xx/tx.c
+++ b/drivers/net/wireless/wl12xx/tx.c
@@ -153,6 +153,9 @@ u8 wl1271_tx_get_hlid(struct wl1271 *wl, struct sk_buff *skb)
 	} else {
 		struct ieee80211_hdr *hdr;
 
+		if (!test_bit(WL1271_FLAG_AP_STARTED, &wl->flags))
+			return wl->system_hlid;
+
 		hdr = (struct ieee80211_hdr *)skb->data;
 		if (ieee80211_is_mgmt(hdr->frame_control))
 			return wl->ap_global_hlid;
-- 
1.7.0.4

